app/controllers/Checkout.scala

```diff
@@ -98,56 +98,55 @@ object Checkout extends Controller with LazyLogging with ActivityTracking with C
     val requestData = SubscriptionRequestData(ProxiedIP.getIP(request))
     val checkoutResult = checkoutService.processSubscription(formData, idUserOpt, requestData)
 
-    checkoutResult.map { result =>
-      result match {
-        case e@CheckoutSuccess(salesforceMember, userIdData, subscribeResult, validPromoCode) => {
-          val productData = Seq(
-            SessionKeys.SubsName -> subscribeResult.subscriptionName,
-            SessionKeys.RatePlanId -> formData.productRatePlanId.get
-          )
-
-          val userSessionFields = userIdData match {
-            case GuestUser(UserId(userId), IdentityToken(token)) =>
-              Seq(SessionKeys.UserId -> userId,
-                SessionKeys.IdentityGuestPasswordSettingToken -> token)
+    checkoutResult.map {
+      case e@CheckoutSuccess(salesforceMember, userIdData, subscribeResult, validPromoCode) => {
+        val productData = Seq(
+          SubsName -> subscribeResult.subscriptionName,
+          RatePlanId -> formData.productRatePlanId.get,
+          SessionKeys.Currency -> formData.personalData.currency.toString
+        )
 
-            case _ => Seq()
-          }
+        val userSessionFields = userIdData match {
+          case GuestUser(UserId(userId), IdentityToken(token)) =>
+            Seq(SessionKeys.UserId -> userId,
+              IdentityGuestPasswordSettingToken -> token)
 
-          val appliedPromoCode = validPromoCode.fold(Seq.empty[(String, String)])(validPromoCode => Seq(SessionKeys.AppliedPromoCode -> validPromoCode.get))
+          case _ => Seq()
+        }
 
-          val session = (productData ++ userSessionFields ++ appliedPromoCode).foldLeft(request.session) {
-            _ + _
-          }
+        val appliedPromoCode = validPromoCode.fold(Seq.empty[(String, String)])(validPromoCode => Seq(AppliedPromoCode -> validPromoCode.get))
 
-          catalog.find(formData.productRatePlanId).foreach { plan =>
-            trackAnon(SubscriptionRegistrationActivity(MemberData(e, formData, plan.billingPeriod)))
-          }
+        val session = (productData ++ userSessionFields ++ appliedPromoCode).foldLeft(request.session.-(AppliedPromoCode)) {
```

Comment:
the -(AppliedPromoCode) bit is key so that a promotion isn't kept between 2 sequential payment flows.